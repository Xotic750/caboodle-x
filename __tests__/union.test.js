import {union} from '../index';

describe('union', () => {
  let args;

  beforeEach(() => {
    args = (function() {
      return arguments;
    })(1, 2, 3);
  });

  it('is a function', () => {
    expect(typeof union).toBe('function');
  });

  it('should throw when array is null or undefined', () => {
    expect(() => {
      union(undefined);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      union(null);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      union([], undefined);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      union([], null);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should return an empty array with n0 arguments', () => {
    expect(union()).toStrictEqual([]);
  });

  it('should remain unchanged with only 1 argument', () => {
    expect(union([1, 2, 3])).toStrictEqual([1, 2, 3]);
  });

  it('should return the union of the given arrays', () => {
    const actual = union([1, 3, 2], [5, 2, 1, 4], [2, 1]);
    expect(actual).toStrictEqual([1, 3, 2, 5, 4]);
  });

  it('should not flatten nested arrays', () => {
    const actual = union([1, 3, 2], [1, [5]], [2, [4]]);
    expect(actual).toStrictEqual([1, 3, 2, [5], [4]]);
  });

  it('should work with `arguments` objects', () => {
    const array = [0];

    expect(union(array, args)).toStrictEqual([0, 1, 2, 3]);
  });
});
