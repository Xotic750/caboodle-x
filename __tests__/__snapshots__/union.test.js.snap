// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`union should throw when array is null or undefined 1`] = `"Cannot call method on null or undefined"`;

exports[`union should throw when array is null or undefined 2`] = `"Cannot call method on null or undefined"`;

exports[`union should throw when array is null or undefined 3`] = `"Cannot call method on null or undefined"`;

exports[`union should throw when array is null or undefined 4`] = `"Cannot call method on null or undefined"`;
